### DISCUSSION TITLE
Combobox autocomplete

### DISCUSSION THREAD

**Discussion by vignesh-sivaprakasam** [twilio-labs/paste#3477] [Q&A] [answered]
Hi team,
     We are using Combobox with autocomplete for one of our use case where we want to search in the list of option available and select. Let's say if the user doesn't select any option, then onBlur we want to show the value with previously selected one. This seems like a common use case. Is there any reason why we are not providing this option by default ? 
     
Here in this example I selected `Deb Never` and onBlur I want the value to be selected. 

https://github.com/twilio-labs/paste/assets/35305421/a844555f-7b34-4d18-87b7-64d9950383ad

**Comment by TheSisb** [comment, marked as answer]
Hi @vignesh-sivaprakasam,

You can achieve your desired functionality by [using the state hook provided](https://paste.twilio.design/components/combobox#usecombobox-state-hook).

[Here's an example in Codesandbox.](https://codesandbox.io/s/combobox-restore-last-selection-6482pk)

**Reply by serifluous** [reply]
Hey @vignesh-sivaprakasam ! Can you share your use case? For this interaction, you'll want to make sure the customer never needs to clear the field, or you'll have to provide a separate button to let them do so.

**Reply by vignesh-sivaprakasam** [reply]
Hi @serifluous , In our pages we have timezone combobox, user will have a default timezone. If the user want to change to any other timezone, then the user can filter through the list of timezones available and then select the desired one. We wont allow the user to clear the selection.

**Reply by vignesh-sivaprakasam** [reply]
Hi @TheSisb , Thanks for sharing the codesandbox link. We already implemented it using the `useCombobox` hook. I felt like this is a common usecase and can this be provided by the Combobox component itself like `resetToLastSelected` prop. If we set it as `true` then the combobox will take care of resetting the value.

**Reply by TheSisb** [reply]
Hi @vignesh-sivaprakasam,

Not sure I agree that it's common enough to be a prop, as it's the first time it's been requested in 3 years. I do appreciate the suggestion though, and will keep it in mind if this becomes a common suggestion.
